src/transformers/modeling_utils.py

```diff
@@ -1667,15 +1742,19 @@ def tie_encoder_to_decoder_recursively(
             module_name: str,
             uninitialized_encoder_weights: List[str],
             depth=0,
+            total_decoder_name="",
```

Comment:
This is important since the `module_name` is a generic name, and `encoder_name` and `decoder_name` can differ ( when there's a ignored cross_attn layer in the tying)